src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/TaskAutoAssignmentService.java

```diff
@@ -119,14 +125,22 @@ public boolean checkAssigneeIsStillValid(TaskResource taskResource, String assig
         List<RoleAssignment> orderedRoleAssignments = orderRoleAssignments(rolesList, roleAssignments);
         return orderedRoleAssignments.stream()
             .filter(roleAssignment -> {
-                TaskRoleResource taskRoleResource = roleResourceMap.get(roleAssignment.getRoleName());
+                final TaskRoleResource taskRoleResource = roleResourceMap.get(roleAssignment.getRoleName());
+
                 if (taskRoleResource == null
-                    || taskRoleResource.getAuthorizations() == null
-                    || taskRoleResource.getAuthorizations().length == 0) {
-                    return false;
-                } else if (roleAssignment.getAuthorisations() == null
-                           || roleAssignment.getAuthorisations().isEmpty()) {
+                    || !taskRoleResource.getAutoAssignable()
+                    || taskRoleResource.getAutoAssignable()
+                        && taskRoleResource.getAuthorizations() != null
+                        && taskRoleResource.getAuthorizations().length > 0
+                        && !findMatchingRoleAssignment(taskRoleResource, roleAssignment)) {
                     return false;
+                } else if (taskRoleResource.getAutoAssignable()
+                            && taskRoleResource.getAuthorizations() != null
+                            && taskRoleResource.getAuthorizations().length == 0) {
+                    return true;
+                } else if (taskRoleResource.getAutoAssignable()
+                           && taskRoleResource.getAuthorizations() == null) {
+                    return true;
                 } else {
                     return findMatchingRoleAssignment(taskRoleResource, roleAssignment);
                 }
```

Comment:
I would look into trying to combine some of these if statements as others in terms of performance you would be accessing the same value multiple times which is not a major issue and sometimes for readability could be okay.

The other thing I would suggest is maybe wrapping this filtering in its own method so it reduces the complexity and can make it easier to maintain and read

Reply:
That is done